File: pkg/matreshka/resources/grpc.go
package resources

const GrpcResourceName = "grpc"

type GRPC struct {
	Name `yaml:"resource_name" env:"-"`

	ConnectionString string `yaml:"connection_string"`
	Module           string `yaml:"module"`
}

func NewGRPC(n Name) Resource {
	return &GRPC{
		Name: n,
	}
}

func (g *GRPC) GetType() string {
	return GrpcResourceName
}

File: pkg/web/Matreshka-UI/src/models/configs/verv/Resources/grpc.ts
import {Node} from "@vervstack/matreshka";
import {GrpcClient} from "@/models/configs/verv/Resources/Resource.ts";
import {extractStringValue} from "@/models/shared/common.ts";

export function mapGrpc(root: Node): GrpcClient {
    if (!root.name) {
        throw {message: 'Can\'t parse grpc client config'}
    }

    const grpcClient = new  GrpcClient(root.name.slice(root.name.indexOf('GRPC')).toLowerCase())

    root.innerNodes?.map(
        (n) => {
            if (!n.name || !root.name) {
                return
            }

            const fieldName = n.name.slice(root.name.length + 1)

            switch (fieldName) {
                case 'CONNECTION-STRING':
                    grpcClient.connectionString = extractStringValue(n)
                    break
                case 'MODULE':
                    grpcClient.module = extractStringValue(n)
                    break
            }
        }
    )

    return grpcClient
}
